### PR TITLE
Refactored Personality Generation and Education Level Assignment.

### LINKED ISSUE #4681: Generating personalities on older saves gives incorrect names
### Environment
16:53:20,669 INFO  [mekhq.MekHQ] {main}
mekhq.MekHQ.initializeLogging(MekHQ.java:315) - Starting MekHQ v0.50.0-SNAPSHOT
    Build Date: 2024-08-17T01:16:16.522116870
    Today: 2024-08-20
    Origin Project: MekHQ
    Java Vendor: Eclipse Adoptium
    Java Version: 17.0.11
    Platform: Windows 10 10.0 (amd64)
    System Locale: en_AU
    Total memory available to MekHQ: 4 GB
    MM Code Revision: 5d74b455d1bfe3c55797e4e3daac3bd0168ca828
    MML Code Revision: fc321e451b068ba6292e43ecc1f6ce80345d79cb
    MHQ Code Revision: b3e0e4457a766489e5545398116ef0d341862003
Nightly 1586

### Description
I have been continuing campaigns from older saves (some older than .49.2), wondering what will break the save.  Enabling the new personality mod saw older characters all with "none" attributes, that also had "average" IQ.  I selected them, and regenerated personalities. Scrolling through, I found a mismatch in some characters names, and what was displayed in their "about".
nope

This issue did not appear in a new game.
As a funny side note, I've got a mech pilot callsign "Squid" whose profile says "hugs others". That works.

### Files
Save file:
[The Red Dragons30360723.cpnx.gz](https://github.com/user-attachments/files/16685536/The.Red.Dragons30360723.cpnx.gz)
Custom files:
[The Red Dragons.zip](https://github.com/user-attachments/files/16685542/The.Red.Dragons.zip)
[ranks.zip](https://github.com/user-attachments/files/16685551/ranks.zip)

### PR DESCRIPTION
**Pull request by IllianiCBT**
Removed duplicate personality generation calls across various classes. Centralized education level assignment and personality generation logic.

### Closes #4681

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/procreation/AbstractProcreation.java [hunk 1]
```diff
@@ -136,7 +135,7 @@ protected int determineNumberOfBabies(final int multiplePregnancyOccurrences) {
     /**
      * This method determines the duration for a pregnancy, with a variance determined through a
      * Gaussian distribution with a maximum spread of approximately six weeks.
-     *
+     * <p>
```
Comment (Sleet01) [inline MekHQ/src/mekhq/campaign/personnel/procreation/AbstractProcreation.java:138]: Is this tag necessary in a comment?

**Review by Sleet01 — APPROVED** [review]
LGTM.  I'm not aware of how the `<p>` tag might be useful in comments, but it's innocuous enough.

**Comment by IllianiCBT** [comment]
The <p> tag prevents IDEA from whining at all the line breaks we have in our java docs. In theory, it also preserves that line break when viewing the doc, allowing us to break up large blocks of text in some of our more verbose method descriptions.

File: MekHQ/src/mekhq/campaign/personnel/procreation/AbstractProcreation.java (hunk 1 above)
Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/personnel/procreation/AbstractProcreation.java:138]: I replied below :)

**Pull request merged**